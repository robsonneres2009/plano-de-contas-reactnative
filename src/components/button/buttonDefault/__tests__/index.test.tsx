import { render, screen, fireEvent } from '@testing-library/react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { Button } from '../..';

describe('Button.Default', () => {
  const mockFn = jest.fn();

  it('Render component primary', () => {
    const tree = renderer
      .create(<Button.Default title="Teste" onPress={mockFn} />)
      .toJSON();
    expect(tree?.children?.length).toBe(1);
  });

  it('Render component secundary', () => {
    const tree = renderer
      .create(<Button.Default title="Teste" onPress={mockFn} secundary />)
      .toJSON();
    expect(tree?.children?.length).toBe(1);
  });

  it('Render component', () => {
    render(<Button.Default title="Teste" onPress={mockFn} />);
    fireEvent.press(screen.getByText('Teste'));
    expect(mockFn).toHaveBeenCalled();
  });

  it('Render component button default', () => {
    const tree = renderer
      .create(<Button.Default title="Teste" onPress={mockFn} secundary />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
