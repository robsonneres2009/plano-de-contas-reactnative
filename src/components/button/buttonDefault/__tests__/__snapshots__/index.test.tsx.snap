// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.Default Render component button default 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ffffff",
        "borderBottomLeftRadius": 24,
        "borderBottomRightRadius": 24,
        "borderTopLeftRadius": 24,
        "borderTopRightRadius": 24,
        "color": ""#fff"",
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  >
    <Text
      secundary={true}
      style={
        {
          "color": "#FF6680",
          "fontSize": 16,
        }
      }
    >
      Teste
    </Text>
  </View>
</View>
`;
